contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -296,6 +298,34 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 		}
 
 		witnessUpdates = preimageSubscription.WitnessUpdates
+
+		// Also subscribe to the htlc interceptor in case a settle comes
+		// in from that direction.
+		interceptorUpdates = make(chan lntypes.Preimage, 1)
+
+		packet := &InterceptedPacket{
```

Comment:
Would've thought that we would've been able to keep the existing `NotifyExitHtlc` "upcall" and thread this information through the invoice registry instead. In theory this would enable us to keep a single execution path from the PoV of the resolvers...🤔

Reply:
With the htlc interceptor, it is possible to "short cut" a forward by suppling the preimage. This preimage isn't available in the invoice registry, because the node isn't the exit hop. How do you envision this exactly?

Reply:
What I mean is that the processing on the resolver side is more or less the same, so we can add a layer of indirection to keep awareness of the higher layers minimal from the resolver's PoV. At the end of the day, it just needs a single thing it calls to get notified if it can obtain the pre-image in time or not. As is the diff pulls a lot of logic into the resolver level that doesn't necessarily need to exist at that level. If we make the current `NotifyExitHtlc` up call into a slim interface, then we can have an implementation that routes to/from the resolver or the invoice registry depending on the context.

Reply:
Ok understood. I changed the code so that the interceptor pushes a received preimage into the beacon. This eliminates the additional channel to listen on in the resolver.

The interceptor still needs to be notified in the resolver, guess there's not really a way around that.

Reply:
> Ok understood. I changed the code so that the interceptor pushes a received preimage into the beacon.

Nice this change should've likely been in place all along for the sake of consistency here. 

> The interceptor still needs to be notified in the resolver, guess there's not really a way around that.

I don't understand why that's the case, left another comment explaining things a bit more.